pdns/fstrm_logger.cc

```diff
@@ -0,0 +1,110 @@
+#include <unistd.h>
+#include "fstrm_logger.hh"
+#include "config.h"
+#ifdef PDNS_CONFIG_ARGS
+#include "logger.hh"
+#define WE_ARE_RECURSOR
+#else
+#include "dolog.hh"
+#endif
+#include <sys/un.h>
+
+#define DNSTAP_CONTENT_TYPE		"protobuf:dnstap.Dnstap"
+
+FrameStreamLogger::FrameStreamLogger(const std::string socket_path): socket_path(socket_path)
+{
+  struct sockaddr_un local;
+  local.sun_family = AF_UNIX;
+  strcpy(local.sun_path, socket_path.c_str());
+
+  fwopt = fstrm_writer_options_init();
```

Comment:
It looks like it can fail and return `nullptr`, please add a check for that case